### ISSUE TITLE
cookiecutter should only process values starting with {{cookiecutter.}}

### ISSUE THREAD

**Issue by professorhaseeb** [cookiecutter/cookiecutter#1624]
## Issue

There are many projects that use {{SOMETHING}} tags for multiple scenarios and cookiecutter processing all tags give an error

```
Unable to create file: filename.ext
Error message: 'SOMETHING' is undefined
```

## Proposed Soluton

cookiecutter should only process values starting with {{cookiecutter.}}

so the rest of the tags can be handled by their respective handlers.

**Comment by professorhaseeb** [comment]
## version information

```
$ pip show cookiecutter
Name: cookiecutter
Version: 1.7.3
Summary: A command-line utility that creates projects from project templates, e.g. creating a Python package project from a Python package project template.
Home-page: https://github.com/cookiecutter/cookiecutter
Author: Audrey Roy
Author-email: audreyr@gmail.com
License: BSD
Location: __MY_PATH__
Requires: python-slugify, poyo, six, requests, binaryornot, click, jinja2-time, Jinja2
```

**Comment by zhongdai** [comment]
@oparaskos Please check out this section regards removing empty lines https://jinja.palletsprojects.com/en/3.0.x/templates/#whitespace-control

**Comment by teebu** [comment]
This draws unsightly extra hash tags where `#{% raw %}` -> `#` was.

**Comment by tylermilner** [comment]
It would be nice to have a better solution for this. I'm running into the same issue with Cookiecutter attempting to render the `${{ context.variable }}` syntax that's also used in GitHub Actions Workflows.

**Problem**
Leaving things unchanged, this block of code in my template directory causes Cookiecutter to fail:
```YAML
concurrency:
  # Cancel any in-progress runs for this workflow/branch combination
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true
```

Results in Cookiecutter runtime error:
```
Unable to create file '.github/workflows/test.yml'
Error message: 'github' is undefined
```

**Solution 1** (❌) - commented `{% raw %}` tags
Adding the `#{% raw %}` comments allows the template to be generated again (✅):
```YAML
concurrency:
  # Cancel any in-progress runs for this workflow/branch combination
  #{% raw %}
  group: ${{ github.workflow }}-${{ github.ref }}
  #{% endraw %}
  cancel-in-progress: true
```

However, this leaves behind empty comment lines in the generated template (❌):
```YAML
concurrency:
  # Cancel any in-progress runs for this workflow/branch combination
  #
  group: ${{ github.workflow }}-${{ github.ref }}
  #
  cancel-in-progress: true
```

**Solution 2** (❌) - uncommented `{% raw %}` tags
Removing the comments and using the `{% raw %}` tags, along with a trailing minus sign `-` does allow for us to get back to a clean generated template (✅):
```YAML
concurrency:
  # Cancel any in-progress runs for this workflow/branch combination
  {% raw -%}
  group: ${{ github.workflow }}-${{ github.ref }}
  {% endraw -%}
  cancel-in-progress: true
```

Renders (✅):
```YAML
concurrency:
  # Cancel any in-progress runs for this workflow/branch combination
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true
```

However this breaks the editing experience when using the VS Code GitHub Actions plugin that provides syntax highlighting/checking of your GHA workflow YAML (❌):
![image](https://github.com/cookiecutter/cookiecutter/assets/1720304/5671e66d-94ce-4f42-b83c-d3215e9b6b07)

**Solution 3** (❌) - inline curly brace `{{` `}}` escaping
Another option that produces identical output to Solution 2 is to use inline curly brace `{{` `}}` escaping (✅):
```YAML
concurrency:
  # Cancel any in-progress runs for this workflow/branch combination
  group: ${{ '{{' }} github.workflow {{ '}}' }}-${{ '{{' }} github.ref {{ '}}' }}
  cancel-in-progress: true
```

Renders (✅):
```YAML
concurrency:
  # Cancel any in-progress runs for this workflow/branch combination
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true
```

No more GitHub Actions plugin syntax issues either (✅):
![image](https://github.com/cookiecutter/cookiecutter/assets/1720304/630e1b90-98f6-434a-be65-49a7be87ae9e)

However, it's a bit of an eyesore and probably not be obvious to new template maintainers what is going on (❌).

**Conclusion**
While there is also the [`_copy_without_render` key](https://cookiecutter.readthedocs.io/en/stable/advanced/copy_without_render.html) that can be added to `cookiecutter.json`, it appears to only work with whole files. This doesn't address the situation where I might want to use Cookiecutter variable substitution in my workflow file while also not needing to work around Cookiecutter trying to render the `{{ github.x }}` syntax.

It would be great if we had a solution like the `_copy_without_render` key, but for text patterns within a file or even double curly braces "contexts" that could be ignored (e.g. **github** in `{{ github.x }}`).

**Comment by lxkaka0736** [comment]
+1
